infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java

```diff
@@ -133,6 +154,48 @@ void prepare(boolean markManaged, boolean canCreate) throws InfrastructureExcept
       namespace = create(name, client);
     }
 
+    if (annotations != null) {
+      KubernetesObjectUtil.putAnnotations(namespace.getMetadata(), annotations);
+    }
+    if (labels != null) {
+      KubernetesObjectUtil.putLabels(namespace.getMetadata(), labels);
+    }
+    if (annotations != null || labels != null) {
+      update(namespace, client);
+    }
+
+    if (resourceQuota != null) {
+      client
+          .resourceQuotas()
+          .inNamespace(name)
+          .createOrReplaceWithNew()
+          .withNewMetadata()
+          .withName("limits")
+          .withNamespace(name)
+          .endMetadata()
+          .withNewSpec()
+          .addToHard(resourceQuota)
+          .endSpec()
+          .done();
+
+      client
+          .limitRanges()
+          .inNamespace(name)
+          .createOrReplaceWithNew()
+          .withNewMetadata()
+          .withName("defaults")
+          .withNamespace(name)
+          .endMetadata()
+          .withNewSpec()
+          .addNewLimit()
+          .withType("Container")
+          .addToDefault(limitRangeLimit)
+          .addToDefaultRequest(limitRangeRequest)
+          .endLimit()
+          .endSpec()
+          .done();
```

Comment:
Why should `LimitRange` creation be dependent on `resourceQuota` being defined?

Reply:
When a ResourceQuota is in place it can stop pods that doesn't have limits or requests set.  This LimitRange is meant to supply the defaults to avoid that problem.